File: P12804.cpp
#include <algorithm>
#include <iostream>
#include <vector>
#define int long long
signed main(void){
	std::ios::sync_with_stdio(false);
	std::cin.tie(nullptr);	
	int t;std::cin >> t;
	while(t--){
		int n ;std::cin >> n;
		std::vector<int> arr(n);
		for(int i = 0;i < static_cast<int>(arr.size());i++) std::cin >> arr[i];
		if(n == 1 || n == 2){
			std::cout << 0 << std::endl;
			continue;
		}	
		std::sort(std::begin(arr),std::end(arr),[](const int & a,const int & b){return a < b;});
		int sum = 0;
		std::for_each(std::begin(arr),std::end(arr),[&sum](auto& x){sum += x;});
		bool flag = true;
		for(int i = n - 1;i >= 0;i--){
			sum -= arr[i];
			if(sum > arr[i]){
				std::cout << sum + arr[i] << std::endl;
				flag = false;	
				break;
			}
		}	
		if(flag == true) std::cout << 0 << std::endl;
	}
	return 0;
}

File: P12804.java
import java.util.Arrays;
import java.util.Scanner;
public class P12804{
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
		StringBuilder sb = new StringBuilder(1 << 20);
		while(t-- > 0){
			int n = sc.nextInt();
			long[] arr = new long[n];
			for(int i = 0;i < n;i++) arr[i] = sc.nextInt();
			if(n == 1 || n == 2) {
				sb.append(0).append('\n');
				continue;
			}
			Arrays.sort(arr);			
			long sum = 0;
			for(int i = 0;i < n;i++) sum += arr[i];
			int res = 0;
			boolean flag = true;
			for(int i = n - 1;i >= 0 ;i--){
				sum -= arr[i];
				if(sum > arr[i]){
					sb.append(sum + arr[i]).append('\n');
					flag = false;
					break;		
				}
			}
			if(flag) sb.append(0).append('\n');
		}
		System.out.print(sb);
		sc.close();
	}
}
